tests/dial_nonexisting.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source "$SCRIPT_DIR"/lib/asserts.sh
+
+N1="test_n1"
+N1_SEED=1
+N1_ADDR="/ip6/::1/udp/53137/quic-v1"
+N1_BAD_ADDR="/ip6/::2/udp/53137/quic-v1"
+N1_BAD_PORT="/ip6/::1/udp/53199/quic-v1"
+N2="test_n2"
+N2_SEED=2
+
+echo "Dial nonexisting node test:"
+
+# run daemon
+killall liberum_core &> /dev/null
+cargo run -p liberum_core -- --daemon  &> /dev/null &
+sleep 0.5; # the socket file is created asynchronously and may not be ready yet :))))
```

Comment:
Why not loop, which waits for file to be created?

Reply:
There are a lot of problems with the scripts. Sleep is used here in all the tests and it's not a part of this PR to change the way the tests are generally performed. If keeping the tests in bash is desirable then I'd prefer making a separate issue/PR which will improve them. If we would like to do the tests more professionally then using something like robot framework would be a much better idea. For now a simple sleep works fine.